### PR TITLE
feat: add rate limit status check on FuelMessagePortalV3

### LINKED ISSUE #315: (feat) Hexens audit FUEL7-5: add rate limiter disabler for FuelMessagePortal

### PR DESCRIPTION
**Pull request by DefiCake**
Closes #315

### CODE REVIEW

**Review by viraj124 — APPROVED** [review]

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 1]
```diff
@@ -272,6 +272,6 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
      * variables without shifting down storage in the inheritance chain.
      * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
      */
-    uint256[49] private __gap;
+    uint256[48] private __gap;
```
Comment (viraj124) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:275]: we added 4 storage vars from the v2 version so should we set this accordingly ?

File: packages/solidity-contracts/test/messagesIncomingV3.test.ts [hunk 2]
```diff
@@ -1177,6 +1179,34 @@ describe('FuelMessagePortalV3 - Incoming messages', () => {
...
+
+      const [msgID, msgBlockHeader, blockInRoot, msgInBlock] = generateProof(
+        messageExceedingRateLimit,
+        blockHeaders,
+        prevBlockNodes,
+        blockIds,
+        messageNodes
+      );
+
+      await fuelMessagePortal.disableRateLimit();
```
Comment (PraneshASP) [inline packages/solidity-contracts/test/messagesIncomingV3.test.ts:1195]: I'd add an additional check before disabling rate limit to verify if the relayMessage fails (when rate limit is enabled) to make sure that disabling rate limit has a real effect.

**Review by PraneshASP — COMMENTED** [review]
lgtm. left a minor comment

**Review by viraj124 — DISMISSED** [review]
lgtm left a minor comment too

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol (hunk 1 above)
Comment (DefiCake) [reply packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:275]: Only `rateLimitEnabled` was added from current mainnet implementation at `0x6d67857224f66d7a677f063b861b6bacafb10639`

**Review by viraj124 — APPROVED** [review]

**Review by PraneshASP — APPROVED** [review]
lgtm

**Pull request merged**